src/validation_workflow/docker/validation_docker.py

```diff
@@ -73,12 +109,19 @@ def validation(self) -> bool:
             if (return_code):
                 logging.info('Cluster is running now')
                 logging.info('Checking if cluster is ready for API test every 10 seconds\n\n')
-                while True:
+
+                self.max_retry = 20
+                self.retry_count = 0
+                while self.retry_count < self.max_retry:
+                    logging.info(f'sleeping 10sec for retry {self.retry_count + 1}/{self.max_retry}')
+                    time.sleep(10)
                     if self.check_http_request():
                         logging.info('\n\ncluster is now ready for API test\n\n')
                         break
-                    logging.info('sleeping 10sec')
-                    time.sleep(10)
+                    self.retry_count += 1
+                else:
+                    logging.warning(f"Maximum number of retries ({self.max_retry}) reached. Cluster is not ready for API test.")
+                    raise Exception(f"Maximum number of retries ({self.max_retry}) reached. Cluster is not ready for API test.")
```

Comment:
Do we need to warn and raise exception both?

Reply:
mm... I wanted to also show a simple string on the output as well ....but yes,, I can certainly remove it ... let me do that .!! thanks!!